add document for HTMLInputElement.type (#32021)
mdn/content: files/en-us/web/api/htmlinputelement/type/index.md

---
title: "HTMLInputElement: type property"
short-title: type
slug: Web/API/HTMLInputElement/type
page-type: web-api-instance-property
browser-compat: api.HTMLInputElement.type
---

{{ApiRef("HTML DOM")}}

> The **`type`** property of the {{domxref("HTMLInputElement")}} interface is a string that indicates behaviour and rendering type of the {{HTMLElement("input")}} element.

Author: ```suggestion
The **`type`** property of the {{domxref("HTMLInputElement")}} interface is a string that indicates behaviour type of the {{HTMLElement("input")}} element, that is, if it is an input for text, a number, a date, etc.
```

It reflects the [`type`](/en-US/docs/Web/HTML/Element/input#type) attribute of the {{HTMLElement("input")}} element.

## Value

> A string representing type values. Default value is a `text`.

Reviewer: The default value is for the attribute, not for the property.
```suggestion
A string representing the type.
```

Author: I wrote this because some properties written with default values. 

But you are right, defaults are for attributes.

> It's possible values are listed in [input types](/en-US/docs/Web/HTML/Element/input#input_types).

Reviewer: ```suggestion
Its possible values are listed in the attribute's [input types](/en-US/docs/Web/HTML/Element/input#input_types) section.
```

## Examples

### HTML

```html
<input id="input1" type="date" />
```

### JavaScript

```js
const inputElement = document.querySelector("#input1");
console.log(inputElement.type); // Output: "date"
```

## Specifications

{{Specifications}}

## Browser compatibility

{{Compat}}

## See also

> - {{domxref("HTMLTextAreaElement.type")}} property that returns the string "textarea".

Reviewer: ```suggestion
- {{domxref("HTMLTextAreaElement.type")}} property
```

> - {{domxref("HTMLTextAreaElement.type")}} property.

Reviewer: A last NIT, that I'll commit.
```suggestion
- {{domxref("HTMLTextAreaElement.type")}} property
```
